core/liquidity-manager/main.ts

```diff
@@ -2,20 +2,51 @@ import { exit } from 'process';
 import { config } from 'dotenv';
 import { Api } from '../shared/api';
 import { Node } from '../shared/node';
+import { Logger } from '../shared/logger';
+import { Util } from '../shared/util';
+import { LiquidityHelper } from './helpers/liquidity-helper';
+import { WithdrawalHelper } from './helpers/withdrawal-helper';
 
 class App {
   private readonly api: Api;
   private readonly node: Node;
+  private readonly logger: Logger;
+
+  private readonly liquidity: LiquidityHelper;
+  private readonly withdrawal: WithdrawalHelper;
 
   constructor() {
     config();
 
     this.api = new Api();
     this.node = new Node();
+    this.logger = new Logger('Liquidity Manager');
+
+    this.liquidity = new LiquidityHelper(this.api, this.node, this.logger);
+    this.withdrawal = new WithdrawalHelper(this.api, this.node, this.logger);
   }
 
   async run(): Promise<void> {
     await this.node.init();
+
+    while (true) {
+      try {
+        const balance = await this.node.getBalance();
+        const withdrawals = await this.withdrawal.getValidWithdrawals();
+        const masternodes = await this.api.getMasternodes();
+
+        // check liquidity
+        // TODO: debounce?
+        await this.liquidity.checkLiquidity(balance, withdrawals, masternodes);
+
+        // payout
+        await this.withdrawal.payoutWithdrawals(balance, withdrawals);
+      } catch (e) {
+        this.logger.error(`Exception: ${e}`);
+      }
+
+      await Util.sleep(60);
```

Comment:
I think we should wait longer here and not just a minute. Debounce is important otherwise we run into double-execution, regardless of sleep time

Reply:
The operations we possibly execute are:
- Payout withdrawals -> balance changes instantly (and we even wait for the TX to complete)
- Build a masternode -> balance changes instantly (and we even wait for the TX to complete)
- Resign a masternode -> API change is instant

-> IMO the changes should be reflected instantly, even if we would have no sleep time at all. WDYT?

Reply:
aaah wrong thinking on my side. execution will wait until await is fulfilled and if we are currently executing, we are not even executing the sleep.

But in the end. Do we want to check balances and everything every minute? I think every 5 is enough and reduces load on all dependencies

Reply:
Yes, you might be right.